quinn-proto/src/connection/streams/mod.rs

```diff
@@ -156,6 +156,29 @@ impl<'a> RecvStream<'a> {
 
         Ok(())
     }
+
+    /// Check whether this stream has been reset by the peer, returning the reset error code if so
+    pub fn reset(&mut self) -> Result<Option<VarInt>, ClosedStream> {
```

Comment:
I think this is a bad name? It reads too much like an active verb, when the semantics are ostensibly supposed to be more like `wait_for_reset()`?

Reply:
I agree that this name sucks. I wish `wait_for_reset().await` wasn't weird and stuttery, though.

Reply:
Yup. Other people must have run into this...

Reply:
I do have the names like this at my `xwt` codebase - `wait_for_bi` when bidirectional connection is opening for instance. I do not even have the irregular verb there, just a different pattern to follow.
So, I would say `wait_for_[...]().await` is not that bad.

---

`has_reset`?

Reply:
seen_reset/reset_seen/was_reset...

Reply:
Renamed to `received_reset`.